### ISSUE TITLE
Syrinscape soundset titles populate in Foundry playlist but none of them play, soundsets failed to fetch

### ISSUE THREAD

**Issue by TheJoiningGame** [frondeus/fvtt-syrin-control#30]
I downloaded the remote control links (CSV) from the Syrinscape website and uploaded them into the foundry module.  I get a list of sound sets automatically populated in my foundry playlist.  But none of them play
<img width="383" alt="Screen Shot 2022-01-30 at 11 43 12 PM" src="https://user-images.githubusercontent.com/98781458/151878041-ea7ce574-7de0-4a3f-927f-e7e5bd231ccd.png">
.

When I select a titled soundset from the dropdown menu in the playlist, no matter which I select, "No soundset" remains checked.  

![Screen_Shot_2022-01-31_at_11 44 30_AM](https://user-images.githubusercontent.com/98781458/151878647-24278d16-9d80-494f-89f0-c56031dc167e.png)

I also input my auth token into the module settings
but…when I save module settings I get the error message;
Syrincontrol; get Soundsets; failed to fetch
<img width="664" alt="Screen Shot 2022-01-31 at 4 47 57 PM" src="https://user-images.githubusercontent.com/98781458/151878675-e2015543-a30d-4430-9044-71a53f696ddb.png">

**Comment by frondeus** [comment]
Alright, first of all thanks for the contribution.

By the first glance I can see that it tries to fetch soundsets, while it shouldn't. After all, you selected "No - stick to CSV file".
I need to debug it further to understand why it doesn't work, because it shouldn't happen.

Could you send me an extra details that could pinpoint where the bug lives?

 - OS: [e.g. Windows 10, Ubuntu 20.10, macOS Catalina]
 - Browser [e.g. chrome 56, safari 12, firefox 90] (Also could you check if this happens only in one browser?)
 - Foundry VTT version [e.g. 8.2, v9]
 - Module version: [e.g. 0.1.0, 0.2.0, 0.2.1]
 - Does it work in Safety Mode (without other plugins enabled)

Also I'd need probably some logs to exactly know what went wrong, to do so please follow these instructions:

To retrieve logs:
* Chrome: Chrome settings button -> More tools -> Developer tools -> Console tab.
* Firefox: Hamburger button (right top corner) -> More Tools -> Web Developer Tools -> Console tab.
* Internet Explorer: press F12 -> Console on the right.
* Microsoft Edge: Click three dots (upper-right corner) -> More tools -> Developer tools -> Console tab.
* Safari: Safari menu -> Preferences -> Advanced tab -> Show Develop -> Develop menu -> Show Web Inspector -> Console tab.

Reload Foundry, then reproduce steps.
Click & drag to select all text,
Right click > Copy
Paste it into surrounding fences (Three backticks), like this:
````
```
SyrinControl | Loading...
SyrinControl | Loaded
```
````

**Comment by TheJoiningGame** [comment]
Thank you for looking into this.  For starters; 

MacOS Catalina 10.15.7
Chrome v97
Foundry version 9.245
Module version 0.2.1

Sorry, but I think I was misleading.  It is only fetching for soundsets when I attempt to sync through auth token, not the CSV file.  

I'll work on getting you the additional info you requested.

**Comment by frondeus** [comment]
Alright. The auth token is still needed even for CSV method - otherwise SyrinControl would not be authorized to actually command Syrinscape to play mood X or Y.

**Comment by TheJoiningGame** [comment]
I got the same exact result in safety mode.  

About your last request, I'm unclear.  Do you mean you want me to uninstall Foundry and reinstall it? Or just shut the program and reopen the program with Chrome console tab open? (I opened the console tab while restarting Foundry and I don't see any text in the console tab.  I don't think I understand how to retrieve the logs.)

**Comment by frondeus** [comment]
I am sorry, by "Reload Foundry", I meant "Reload/refresh current page" (Usually Ctrl+R works).
You don't have to restart or uninstall foundry. Just refresh the page in the browser with the console opened so the console can catch all the logs.

**Comment by TheJoiningGame** [comment]
'''
test
'''

**Comment by TheJoiningGame** [comment]
```
﻿
foundry.js:3053 Foundry VTT | Rendering SettingsConfig
foundry.js:4660 Foundry VTT | Retrieved and compiled template templates/sidebar/apps/settings-config.html
foundry.js:4660 Foundry VTT | Retrieved and compiled template templates/app-window.html
settings.ts:92 SyrinControl | config 
S.fn.init [div.settings-list, prevObject: S.fn.init(1)]
SettingsConfig.svelte:22 SyrinControl | onCSVSelected 
File {name: 'syrinscape_remote_control_links_James_Green.csv', lastModified: 1643602341000, lastModifiedDate: Sun Jan 30 2022 23:12:21 GMT-0500 (Eastern Standard Time), webkitRelativePath: '', size: 20492826, …}
SettingsConfig.svelte:30 SyrinControl | Loaded "id","status","subcategory","product_or_pack","soundset","name","type","sub_type","genre_players_play_url","genre_players_stop_url","online_player_play_url","online_player_stop_url"
"m:257254","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Wandering the streets","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6V2FuZGVyaW5nIHRoZSBzdHJlZXRz/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6V2FuZGVyaW5nIHRoZSBzdHJlZXRz/stop/","https://syrinscape.com/online/frontend-api/moods/257254/play/","https://syrinscape.com/online/frontend-api/moods/257254/stop/"
"m:257255","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Halls, rooms and passages","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6SGFsbHMsIHJvb21zIGFuZCBwYXNzYWdlcw/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6SGFsbHMsIHJvb21zIGFuZCBwYXNzYWdlcw/stop/","https://syrinscape.com/online/frontend-api/moods/257255/play/","https://syrinscape.com/online/frontend-api/moods/257255/stop/"
"m:257256","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Auril 1st form battle","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6QXVyaWwgMXN0IGZvcm0gYmF0dGxl/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6QXVyaWwgMXN0IGZvcm0gYmF0dGxl/stop/","https://syrinscape.com/online/frontend-api/moods/257256/play/","https://syrinscape.com/online/frontend-api/moods/257256/stop/"
"m:257257","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","(more music)","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6KG1vcmUgbXVzaWMp/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6KG1vcmUgbXVzaWMp/stop/","https://syrinscape.com/online/frontend-api/moods/257257/play/","https://syrinscape.com/online/frontend-api/moods/257257/stop/"
"m:257258","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Auril 2nd form battle","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6QXVyaWwgMm5kIGZvcm0gYmF0dGxl/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6QXVyaWwgMm5kIGZvcm0gYmF0dGxl/stop/","https://syrinscape.com/online/frontend-api/moods/257258/play/","https://syrinscape.com/online/frontend-api/moods/257258/stop/"
"m:257259","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","(more music 2nd)","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6KG1vcmUgbXVzaWMgMm5kKQ/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6KG1vcmUgbXVzaWMgMm5kKQ/stop/","https://syrinscape.com/online/frontend-api/moods/257259/play/","https://syrinscape.com/online/frontend-api/moods/257259/stop/"
"m:257260","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Auril 3rd form battle","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6QXVyaWwgM3JkIGZvcm0gYmF0dGxl/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6QXVyaWwgM3JkIGZvcm0gYmF0dGxl/stop/","https://syrinscape.com/online/frontend-api/moods/257260/play/","https://syrinscape.com/online/frontend-api/moods/257260/stop/"
"m:257261","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","(more music 3rd)","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6KG1vcmUgbXVzaWMgM3JkKQ/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6KG1vcmUgbXVzaWMgM3JkKQ/stop/","https://syrinscape.com/online/frontend-api/moods/257261/play/","https://syrinscape.com/online/frontend-api/moods/257261/stop/"
"m:257262","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Summoning a tarrasque","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6U3VtbW9uaW5nIGEgdGFycmFzcXVl/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6U3VtbW9uaW5nIGEgdGFycmFzcXVl/stop/","https://syrinscape.com/online/frontend-api/moods/257262/play/","https://syrinscape.com/online/frontend-api/moods/257262/stop/"
"m:257265","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Epilogue - perpetual winter","mood","","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6RXBpbG9ndWUgLSBwZXJwZXR1YWwgd2ludGVy/play/","syrinscape-fantasy:moods/cm90Zi1lcGlsb2d1ZTo6RXBpbG9ndWUgLSBwZXJwZXR1YWwgd2ludGVy/stop/","https://syrinscape.com/online/frontend-api/moods/257265/play/","https://syrinscape.com/online/frontend-api/moods/257265/stop/"
"m:257266","OFFICIAL","dungeons-and-dragons","Icewind Dale: Rime of the Frostmaiden Chapter 7 Soundpack","rotf Auril's wrath and Epilogue","Epilogue - summer is coming","mood","","syrinscape-fantas
csv.ts:5 SyrinControl | Control Links syrinscape_remote_control_links_James_Green.csv
SettingsConfig.svelte:27 [Violation] 'load' handler took 906ms
csv.ts:22 SyrinControl  CSV| 
{data: Array(44442)}
csv.ts:69 SyrinControl | Loaded CSV
foundry.js:3053 Foundry VTT | Rendering SettingsConfig
settings.ts:92 SyrinControl | config 
S.fn.init [div.settings-list, prevObject: S.fn.init(1)]
game:1 Access to fetch at 'https://syrinscape.com/online/frontend-api/soundsets?auth_token=8162752…' from origin 'http://74.106.9.29:30000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
raw.ts:90 GET https://syrinscape.com/online/frontend-api/soundsets?auth_token=8162752… net::ERR_FAILED 301
﻿
​```

**Comment by TheJoiningGame** [comment]
<img width="1143" alt="Screen Shot 2022-02-01 at 7 23 36 PM" src="https://user-images.githubusercontent.com/98781458/152073458-bde395e8-9745-4315-b13f-f7039813898b.png">

**Comment by TheJoiningGame** [comment]
This is the result I'm getting in safe mode with the only active module being Syrinscape Control.

**Comment by TheJoiningGame** [comment]
The red "Access to fetch..." text appears when I switch from "Sync by CSV" to "use API".

**Comment by BrewerSeth** [comment]
I am also receiving the same user interface result as @TheJoiningGame. When I click a sound set, it doesn't appear to select and does not populate the "Mood" drop down. See gif:

![ezgif-2-f377cb7f2e](https://user-images.githubusercontent.com/86263756/152411500-5cbd98c9-2bf5-4d25-99c8-ff4c9ed2b7c1.gif)

But I have confirmed that the Auth Token is set up correctly. I can trigger my Syrinscape online using Macros like this:

![Screen Shot 2022-02-03 at 1 07 06 PM](https://user-images.githubusercontent.com/86263756/152411911-28a87c7e-80ae-4a4f-b6a6-044be6124b06.png)

**Comment by frondeus** [comment]
> I am also receiving the same user interface result as @TheJoiningGame. When I click a sound set, it doesn't appear to select and does not populate the "Mood" drop down. See gif:
> 
> ![ezgif-2-f377cb7f2e](https://user-images.githubusercontent.com/86263756/152411500-5cbd98c9-2bf5-4d25-99c8-ff4c9ed2b7c1.gif)
> 
> But I have confirmed that the Auth Token is set up correctly. I can trigger my Syrinscape online using Macros like this:
> 
> ![Screen Shot 2022-02-03 at 1 07 06 PM](https://user-images.githubusercontent.com/86263756/152411911-28a87c7e-80ae-4a4f-b6a6-044be6124b06.png)

Thank you for the gif. I am going to hunt this bug ASAP, but before I start I think this is a good moment to do #18 - Because I rarely use CSV method I do not see these bugs so automatic tests would ensure it never happens again :)

**Comment by 14VTTgame** [comment]
I have the same issue and none of the technical skills needed to solve it. I also am on a Mac (OS Big Sur, 11.6.3) and am using Foundry version 9.245

I followed the CVS installation steps carefully and everything seemed to work until I went to actually pick a soundset. Like the others, I could see all of the available soundsets, but could not choose one. I have been able to install sounds one at a time using macro buttons that link to the online Syrinscape player.

One additional bit of information: There is nothing listed under Foundry's "Ambient Sound Controls."

<img width="580" alt="Screen Shot 2022-02-04 at 12 06 41 PM" src="https://user-images.githubusercontent.com/99038986/152571676-43397ca2-32f0-49d7-97d6-b0b0602fd69c.png">

**Comment by Koboldking12** [comment]
I'm having the same issues. I get the drop down menu but are unable to select anything. How do I resolve this?

<img width="674" alt="Screen Shot 2022-02-09 at 1 43 33 PM" src="https://user-images.githubusercontent.com/99300628/153100242-2c564d4a-848f-43d3-968a-531284dc0412.png">

**Comment by frondeus** [comment]
Working on a solution.

**Comment by bobdigital78** [comment]
I have the same issue and getting this error when selecting my sound set after attaching the CSV file. I also noticed that when I return to the module settings my CSV is no longer showing as attached. My auth token is still there
running on Mac OS Monterey
Foundry v.9
<img width="1422" alt="Screen Shot 2022-02-16 at 7 50 35 PM" src="https://user-images.githubusercontent.com/56169597/154249670-7db69f8b-1b6d-44ae-9590-0d0a639d474f.png">
r

**Comment by frondeus** [comment]
The CSV is not going to stay attached - because SyrinControl does not need it after loading the data.

**Comment by frondeus** [comment]
Hello everyone! 
I have some good news!

While I'm preparing the new, more polished and test-covered version of SyrinControl, the Syrinscape Team finally fixed the CORS issue!

It means you can now try the **Online API without setting the CORS proxy**. It should just work :)

Nevertheless - I know that recently I was a bit silent about the issue, but I was actively working on the new version which should be more reliable. More news soon!

**Comment by karpana** [comment]
@frondeus 

That is excellent news. 
Is there some place we can download this copy to give it a whirl?

I ask, because the latest release on GitHub is from December 29th 2021. 

Let me know. I would love to give this a whirl.

**Comment by karpana** [comment]
oh.... 
I misunderstood that the ***existing*** version would work as-is.
I'll give that a whirl.

Thanks @frondeus 

Much appreciated.

**Comment by karpana** [comment]
@frondeus 

this works fantastically!!
Thank you to you and all your efforts getting this up and running...
And also to the good folks @ Syrinscape for accomodating this!

**Comment by frondeus** [comment]
I just released v0.3.0, the CSV method does no longer exist. If a similar problem occurs in the new version using online API please create a new issue.
